bvm/ballerina-core/src/main/java/org/ballerinalang/util/program/BLangFunctions.java

```diff
@@ -427,13 +430,27 @@ private static void invokeNativeCallableAsync(CallableUnitInfo callableUnitInfo,
     private static void handleError(WorkerExecutionContext ctx) {
         throw new BLangRuntimeException("error: " + BLangVMErrors.getPrintableStackTrace(ctx.getError()));
     }
-    
-    private static WorkerExecutionContext executeWorker(WorkerResponseContext respCtx, 
+
+    private static WorkerExecutionContext executeWorker(WorkerResponseContext respCtx,
+            WorkerExecutionContext parentCtx, int[] argRegs, CallableUnitInfo callableUnitInfo,
+            WorkerInfo workerInfo, WorkerDataIndex wdi, WorkerData initWorkerLocalData,
+            CodeAttributeInfo initWorkerCAI, boolean runInCaller, ObserverContext observerContext) {
+        return executeWorker(respCtx, parentCtx, argRegs, callableUnitInfo, workerInfo, wdi, initWorkerLocalData,
+                initWorkerCAI, runInCaller, observerContext, 0);
+    }
+
+    private static WorkerExecutionContext executeWorker(WorkerResponseContext respCtx,
             WorkerExecutionContext parentCtx, int[] argRegs, CallableUnitInfo callableUnitInfo, 
             WorkerInfo workerInfo, WorkerDataIndex wdi, WorkerData initWorkerLocalData, 
-            CodeAttributeInfo initWorkerCAI, boolean runInCaller, ObserverContext observerContext) {
-        WorkerData workerLocal = BLangVMUtils.createWorkerDataForLocal(workerInfo, parentCtx, argRegs,
-                callableUnitInfo.getParamTypes());
+            CodeAttributeInfo initWorkerCAI, boolean runInCaller, ObserverContext observerContext, int flags) {
+        WorkerData workerLocal;
+        if (FunctionFlags.isCompensate(flags)) {
+            workerLocal = ((CompensationTable) parentCtx.globalProps.get(Constants.COMPENSATION_TABLE))
```

Comment:
This will return the same WorkerData. Isn't it? What is the reason for storing worker data in compensation table. ?